sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -241,7 +241,7 @@ case class NewInstance(
         $javaType ${ev.value} = ${ctx.defaultValue(dataType)};
         if ($argsNonNull) {
           ${ev.value} = $constructorCall;
-          ${ev.isNull} = false;
+          ${ev.isNull} = ${ev.value} == null;
```

Comment:
Since it is null or not will be depended on ev.value no matter the value of `propagateNull`, should we set the `nullable` of `NewInstance` to true. Currently it is assigned from `propagateNull`.